util_test.go

```diff
@@ -18,10 +18,7 @@ func comparePokemon(a, b *Pokemon) bool {
 	} else if a == nil || b == nil {
 		return false
 	}
```

Comment:
Is this case necessary now that we have the `a == b` fallthrough?

Reply:
Nope, removing it